ep0099/ep0099.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package ep0099
+
+import (
+	"errors"
+
+	"periph.io/x/conn/v3/i2c"
+)
+
+var InvalidAddressError = errors.New("Invalid EP-0099 address")
+var InvalidChannelError = errors.New("Invalid EP-0099 channel")
+
+type State byte
```

Comment:
Document
Why not a bool?

Reply:
the idea was to have StateOn and StateOff match the value that would be send to the device that represents the desired state so it is easier to read tests when the assertion to `dev.On` checks if the value recorded was StateOn.

it can be easily updated to bool and Write send 0x00 or 0xFF to the device.

Reply:
On one hand I would have wanted to use gpio.Level but at the same time it doesn't mean exactly the same thing; gpio.High doesn't necessarily mean "On".

The vision is that Dev could embed 4 Relay objects, which then would implement gpio.PinOut https://periph.io/x/conn/v3/gpio#PinOut

The value proposition is that then they are simply pins that can be forwarded to other systems.

pseudo code would look like

```
type Dev struct {
  Relays [4]Relay
}

type Relay struct {
  d*Dev   <- pointer back to the main struct
  state bool
}

type (r *Relay) Out(l gpio.Level) error {
	_, err := d.i2c.Write([]byte{channel, byte(StateOn)})
	r.state = StateOn
	return err
}
```

Then users can do

```
d := ep0099.New(...)
d.Relays[0].Out(gpio.High)
d.Relays[0].Out(gpio.Low)
fmt.Printf("%s\n", d.Relay[0].Read())
```

I don't know if users would see this as good ergonomics. The nice thing is that Relays is an array so the compiler enforces boundary checks for you. I try to make things generic when possible but it doesn't always make sense.

TL;DR: as you see fit.